pkg/agent/controller/egress/egress_controller.go

```diff
@@ -443,6 +500,32 @@ func (c *EgressController) unbindPodEgress(pod, egress string) (string, bool) {
 	return "", false
 }
 
+func (c *EgressController) updateEgressStatus(egress *crdv1a2.Egress, nodeName string) error {
+	if egress.Status.NodeName == nodeName {
+		return nil
+	}
+	klog.V(2).InfoS("Updating Egress status", "Egress", egress.Name, "oldNode", egress.Status.NodeName, "newNode", nodeName)
+	toUpdate := egress.DeepCopy()
+	toUpdate.Status.NodeName = nodeName
+	if _, err := c.crdClient.CrdV1alpha2().Egresses().UpdateStatus(context.TODO(), toUpdate, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("updating Egress %s status error: %v", egress.Name, err)
+	}
+	klog.V(2).InfoS("Updated Egress status", "Egress", egress.Name)
+	metrics.AntreaEgressStatusUpdates.Inc()
+	return nil
+}
+
+// unassignEgressIP unassigned the Egress IP which assigned by agent.
+func (c *EgressController) unassignEgressIP(egressName string) error {
+	if ip, ok := c.ipAssigner.IsAssignedIP(egressName); ok && c.localIPDetector.IsLocalIP(ip) {
+		if err := c.ipAssigner.UnassignEgressIP(ip); err != nil {
```

Comment:
Could you consider my suggestion in https://github.com/antrea-io/antrea/pull/2186#discussion_r662412807 and see we can encapsulate more logic in ipAssigner since it alreay has its own cache mapping egress to egressIP? I think it doesn't need to check `IsLocalIP` as it needs to clean up cache even if IsLocalIP is false.